### PR TITLE
[Console] Fix autocomplete multibyte input support

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | https://github.com/symfony/symfony/issues/34254
| License       | MIT
| Doc PR        | -

Added it just where it's needed.

### CODE REVIEW

File: src/Symfony/Component/Console/Helper/QuestionHelper.php [hunk 1]
```diff
@@ -264,7 +264,7 @@ private function autocomplete(OutputInterface $output, Question $question, $inpu
             } elseif ("\177" === $c) { // Backspace Character
                 if (0 === $numMatches && 0 !== $i) {
                     --$i;
-                    $fullChoice = substr($fullChoice, 0, -1);
+                    $fullChoice = mb_substr($fullChoice, 0, -1);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Helper/QuestionHelper.php]: Using `mb_*` without proving the `$encoding` argument is relying on hidden global state - better not.
Could we instead use the function only when $fullChoice is UTF-8? (using `preg_match('//u', ...)`)?

**Comment by nicolas-grekas** [comment]
Oh, and please add a test case if possible :)

**Comment by fancyweb** [comment]
I used existing helper functions and added a test case.

**Review by chalasr — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Pull request merged**